static/js/components/input/ProductSelector.js

```diff
@@ -162,17 +162,33 @@ export class ProductSelector extends React.Component<Props, State> {
     ) {
       return []
     }
-
+    //Get today's date
+    const todaysDate = new Date()
     if (productType === PRODUCT_TYPE_PROGRAM) {
-      return buildProgramDateOptions(programRuns)
+      return buildProgramDateOptions(
+        programRuns.filter(programRun => {
+          let startDate = null
+          if (programRun.start_date) {
+            startDate = new Date(programRun.start_date)
+          }
+          return startDate && startDate >= todaysDate
+        })
+      )
     } else {
       return buildCourseDateOptions(
-        products.filter(
-          product =>
+        products.filter(product => {
+          let startDate = null
+          if (product.content_object.start_date) {
+            startDate = new Date(product.content_object.start_date)
```

Comment:
If I understood @briangrossman comment on the original ticket, this check should be on enrollment start/end dates not the course run start/end dates.

Reply:
@ahmed-arbisoft the drop is operating over course start/end dates in b2b bulk purchase section.

Reply:
@asadiqbal08 and @ahmed-arbisoft  / @ahmed-belal 

Matching the behavior to be the same as how the course runs are filtered probably makes the most sense. So, if that's what you're doing, that seems like the right approach.

Reply:
Going through the current implementation of metadata tiles on the product page I see that we are using unexpired_runs attribute on the course runs which in turn verify if the course run has not ended yet and the enrollment duration is still applicable. Given that that's the case on product pages, I suggest we change this to incorporate enrollment dates the same as is being done in unexpired_runs property on the products.